tests/failpoints/cases/test_split_region.rs

```diff
@@ -378,8 +378,126 @@ fn test_split_not_to_split_exist_tombstone_region() {
+
+    let r1 = cluster.run_conf_change();
+
+    pd_client.must_add_peer(r1, new_peer(3, 3));
+
+    assert_eq!(r1, 1);
+
+    let before_check_snapshot_1_2_fp = "before_check_snapshot_1_2";
+    fail::cfg(before_check_snapshot_1_2_fp, "pause").unwrap();
+
+    pd_client.must_add_peer(r1, new_peer(2, 2));
+
+    cluster.must_put(b"k1", b"v1");
+    cluster.must_put(b"k2", b"v2");
+
+    let before_check_snapshot_1000_2_fp = "before_check_snapshot_1000_2";
+    fail::cfg(before_check_snapshot_1000_2_fp, "pause").unwrap();
+
+    let region = pd_client.get_region(b"k1").unwrap();
+    cluster.must_split(&region, b"k2");
+    let left = pd_client.get_region(b"k1").unwrap();
+    assert_eq!(left.get_id(), 1000);
+
+    cluster.must_put(b"k11", b"v11");
+
+    // Wait for region 1000 sending heartbeat and snapshot to store 2
+    sleep_ms(200);
+
+    fail::remove(before_check_snapshot_1_2_fp);
+    // peer 2 applied snapshot
+    must_get_equal(&cluster.get_engine(2), b"k1", b"v1");
+
+    fail::remove(before_check_snapshot_1000_2_fp);
+
+    must_get_equal(&cluster.get_engine(2), b"k11", b"v11");
+}
+
+// Test if a peer can be created from splitting when another uninitialied peer with different
+// peer id has been created on this store.
+#[test]
+fn test_split_not_to_split_existing_different_uninitialied_peer() {
+    let mut cluster = new_node_cluster(0, 3);
+    configure_for_merge(&mut cluster);
+    cluster.cfg.raft_store.right_derive_when_split = true;
+    cluster.cfg.raft_store.store_batch_system.max_batch_size = 1;
+    cluster.cfg.raft_store.store_batch_system.pool_size = 2;
+    cluster.cfg.raft_store.apply_batch_system.max_batch_size = 1;
+    cluster.cfg.raft_store.apply_batch_system.pool_size = 2;
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+
+    fail::cfg("on_raft_gc_log_tick", "return()").unwrap();
+    let r1 = cluster.run_conf_change();
+
+    pd_client.must_add_peer(r1, new_peer(3, 3));
+
+    assert_eq!(r1, 1);
+
+    let before_check_snapshot_1_2_fp = "before_check_snapshot_1_2";
+    fail::cfg(before_check_snapshot_1_2_fp, "pause").unwrap();
+
+    pd_client.must_add_peer(r1, new_peer(2, 2));
+
+    cluster.must_put(b"k1", b"v1");
+    cluster.must_put(b"k2", b"v2");
+
+    // Wait for region 1 sending heartbeat and snapshot to store 2
+    sleep_ms(200);
+
+    cluster.add_send_filter(IsolationFilterFactory::new(2));
+
+    let region = pd_client.get_region(b"k1").unwrap();
+    cluster.must_split(&region, b"k2");
+    let left = pd_client.get_region(b"k1").unwrap();
+    assert_eq!(left.get_id(), 1000);
+    let left_peer_2 = find_peer(&left, 2).cloned().unwrap();
+
+    pd_client.must_remove_peer(left.get_id(), left_peer_2);
+    pd_client.must_add_peer(left.get_id(), new_peer(2, 4));
+
+    let before_check_snapshot_1000_2_fp = "before_check_snapshot_1000_2";
+    fail::cfg(before_check_snapshot_1000_2_fp, "pause").unwrap();
```

Comment:
How about setting the fail point before splitting?

Reply:
There is no difference because there is a send filter of store 2.